R/compat-types-check.R

```diff
@@ -103,9 +148,15 @@ check_number <- function(x,
     }
   }
 
+  if (allow_decimal) {
+    what <- "a number"
+  } else {
+    what <- "a round number"
```

Comment:
```suggestion
    what <- "a whole number"
```

Since the function is `check_number_whole()`. This also sounds a little better to me